src/cargo/util/process_builder.rs

```diff
@@ -329,6 +347,53 @@ impl ProcessBuilder {
         }
         command
     }
+
+    /// Converts `ProcessBuilder` into a `std::process::Command` and a rustc style response
+    /// file.  Also handles the jobserver, if present.
+    pub fn build_command_and_response_file(&self) -> CargoResult<(Command, tempfile::TempPath)> {
+        // The rust linker also jumps through similar hoops, although with a different
+        // of response file, which this borrows from.  Some references:
+        // https://github.com/rust-lang/rust/blob/ef92009c1dbe2750f1d24a6619b827721fb49749/src/librustc_codegen_ssa/back/link.rs#L935
+        // https://github.com/rust-lang/rust/blob/ef92009c1dbe2750f1d24a6619b827721fb49749/src/librustc_codegen_ssa/back/command.rs#L135
+
+        let mut command = Command::new(&self.program);
+        if let Some(cwd) = self.get_cwd() {
+            command.current_dir(cwd);
+        }
+        // cmd.exe can handle up to 8k work of args, this leaves some headroom if using a .cmd rustc wrapper.
+        let mut cmd_remaining: usize = 1024 * 6;
```

Comment:
I think it's fine to avoid this heuristic and simply pass everything through a response file once we've failed once.

Reply:
A RUSTC_WRAPPER is likely to fail to understand response files.  This:
```cmd
some_rustc_wrapper path/to/rustc.exe --flag --flag --flag ... @remaining_args
```
Is likely to work, while this:

```cmd
some_rustc_wrapper @all_args_including_path_to_rustc_exe
```
Will almost certainly break.  Perhaps that's fine - or even preferred?  Loudly breaking instead of silently failing to recognize/modify flags found in `@remaining_args` ?

Reply:
Er yes that's a good point, could the arguments to rustc be handled a bit differently in that case so only rustc's arguments get into the `@`-file?

This may be a bit too low-level of a location to put the insertion of the `@`-file